test/test_Gaussian/main.cpp

```diff
@@ -24,7 +24,7 @@ int main()
             std::cout << "Caught error.\n" << e.what() << std::endl;
         }
 
-        if (!catch_exception)
+        if (catch_exception)
```

Comment:
Please remove the `catch_expetion` variable and move the `return EXIT_FAILURE` inside the `catch` block.